dask_cuda/proxy_object.py

```diff
@@ -0,0 +1,483 @@
+
+    def __rmod__(self, other):
+        return other % self._obj_pxy_deserialize()
+
+    def __rdivmod__(self, other):
+        return divmod(other, self._obj_pxy_deserialize())
+
+    def __rpow__(self, other, *args):
+        return pow(other, self._obj_pxy_deserialize(), *args)
+
+    def __rlshift__(self, other):
+        return other << self._obj_pxy_deserialize()
+
+    def __rrshift__(self, other):
+        return other >> self._obj_pxy_deserialize()
+
+    def __rand__(self, other):
+        return other & self._obj_pxy_deserialize()
+
+    def __rxor__(self, other):
+        return other ^ self._obj_pxy_deserialize()
+
+    def __ror__(self, other):
+        return other | self._obj_pxy_deserialize()
+
+    def __iadd__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied += other
+        return self
+
+    def __isub__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied -= other
+        return self
+
+    def __imul__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied *= other
+        return self
+
+    def __itruediv__(self, other):
+        with self._obj_pxy_lock:
+            proxied = self._obj_pxy_deserialize()
+            self._obj_pxy["obj"] = operator.itruediv(proxied, other)
+        return self
+
+    def __ifloordiv__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied //= other
+        return self
+
+    def __imod__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied %= other
+        return self
+
+    def __ipow__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied **= other
+        return self
+
+    def __ilshift__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied <<= other
+        return self
+
+    def __irshift__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied >>= other
+        return self
+
+    def __iand__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied &= other
+        return self
+
+    def __ixor__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied ^= other
+        return self
+
+    def __ior__(self, other):
+        proxied = self._obj_pxy_deserialize()
+        proxied |= other
+        return self
+
+    def __neg__(self):
+        return -self._obj_pxy_deserialize()
+
+    def __pos__(self):
+        return +self._obj_pxy_deserialize()
+
+    def __abs__(self):
+        return abs(self._obj_pxy_deserialize())
+
+    def __invert__(self):
+        return ~self._obj_pxy_deserialize()
+
+    def __int__(self):
+        return int(self._obj_pxy_deserialize())
+
+    def __float__(self):
+        return float(self._obj_pxy_deserialize())
+
+    def __complex__(self):
+        return complex(self._obj_pxy_deserialize())
+
+    def __index__(self):
+        return operator.index(self._obj_pxy_deserialize())
```

Comment:
Some are needed for `ProxyObject` to pass-through cudf and tpcx-bb workflows but many of them are just there for completeness.

Reply:
I see, given they're already there I won't suggest we remove them, but for future reference I think we can try to keep code shorter instead of solving all possible cases that are probably unnecessary, this helps with future maintainability.

Reply:
I agree but since `ProxyObject` is exposed to the end user, I think it is reasonable to support most common operations. Say a user writes a Dask tasks that uses Numpy arrays, the user should be able to use most, if not all, Numpy operations.